components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -0,0 +1,169 @@
+                 aer_tau_sw_src("aer_tau_sw_src",ncols,nswbands,pressure_state.nlevs),
+                 aer_tau_lw_src("aer_tau_lw_src",ncols,nlwbands,pressure_state.nlevs);
+
+  using ExeSpace = typename KT::ExeSpace;
+  const Int nk_pack = ekat::npack<Spack>(nlevs);
+  const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(ncols, nk_pack);
+  // SPA Main loop
+  Kokkos::parallel_for(
+    "spa main loop",
+    policy,
+    KOKKOS_LAMBDA(const MemberType& team) {
+
+      const Int i = team.league_rank();
+
+      // Get single-column subviews of all inputs
+      const auto& ps_beg_sub                 = pressure_state.ps_this_month(i);
+      const auto& ps_end_sub                 = pressure_state.ps_next_month(i);
+      const auto& pmid_sub                   = ekat::subview(pressure_state.pmid, i);
+      const auto& p_src_sub                  = ekat::subview(p_src, i);
+
+      const auto& ccn3_beg_sub               = ekat::subview(data_beg.CCN3, i);
+      const auto& ccn3_end_sub               = ekat::subview(data_end.CCN3, i);
+      const auto& ccn3_src_sub               = ekat::subview(ccn3_src, i);
+
+      // First Step: Horizontal Interpolation if needed - Skip for Now
+  
+      // Second Step: Temporal Interpolation
+      auto slope = (t_now-t_beg)/t_len;
+      /* Determine PS for the source data at this time */
+      auto ps_src  =  ps_beg_sub + slope * (ps_end_sub-ps_beg_sub);
+      /* Reconstruct the vertical pressure profile for the data and time interpolation
+       * of the data */
+      {
+      using C = scream::physics::Constants<Real>;
+      static constexpr auto P0 = C::P0;
+      const Int nk_pack = ekat::npack<Spack>(pressure_state.nlevs);
+      Kokkos::parallel_for(
+        Kokkos::TeamThreadRange(team, nk_pack), [&] (Int k) {
+          p_src_sub(k)    = ps_src * pressure_state.hybm(k) + P0 * pressure_state.hyam(k);
+          ccn3_src_sub(k) = ccn3_beg_sub(k) + slope * (ccn3_end_sub(k) - ccn3_beg_sub(k));
+      });
+      team.team_barrier();
+      Kokkos::parallel_for(
+        nswbands, [&] (int n) {
+        const auto& aer_g_sw_beg_sub           = ekat::subview(data_beg.AER_G_SW, i, n);
+        const auto& aer_g_sw_end_sub           = ekat::subview(data_end.AER_G_SW, i, n);
+        const auto& aer_g_sw_src_sub           = ekat::subview(aer_g_sw_src, i, n);
+
+        const auto& aer_ssa_sw_beg_sub         = ekat::subview(data_beg.AER_SSA_SW, i, n);
+        const auto& aer_ssa_sw_end_sub         = ekat::subview(data_end.AER_SSA_SW, i, n);
+        const auto& aer_ssa_sw_src_sub         = ekat::subview(aer_ssa_sw_src, i, n);
+  
+        const auto& aer_tau_sw_beg_sub         = ekat::subview(data_beg.AER_TAU_SW, i, n);
+        const auto& aer_tau_sw_end_sub         = ekat::subview(data_end.AER_TAU_SW, i, n);
+        const auto& aer_tau_sw_src_sub         = ekat::subview(aer_tau_sw_src, i, n);
+
+        Kokkos::parallel_for(
```

Comment:
Since this is nested in a team loop, this should be ThreadVectorRange. Or perhaps the outer loop is meant to be a kernel launch.